### PR TITLE
Add support for loading users `~/.{irb/pry}rc`

### LINKED ISSUE #119: Consider sourcing ~/.pryrc
Would be nice if a user's `~/.pryrc` and/or `~/.irbrc` preferences could be included in the docker build.

### PR DESCRIPTION
**Pull request by johnallen3d**
Resolves #119

### CODE REVIEW

File: lib/nib/history.rb [hunk 1]
```diff
@@ -1,32 +1,62 @@
...
+      else
+        Nib.load_config(:shell, type)
+      end
+    end
+
+    def config_file
+      @config_file ||= File.open("./tmp/#{type}", 'w+') do |file|
+        file.write(config)
+        file.write(history_command)
+        file
```
Comment (blankenshipz) [inline lib/nib/history.rb]: 👏

**Comment by johnallen3d** [comment]
👀 @blankenshipz, @binarymason

File: lib/nib/history.rb [hunk 2]
```diff
@@ -1,32 +1,62 @@
-module Nib::History
-  IRBRC = <<-'IRB'.freeze
```
Comment (binarymason) [inline lib/nib/history.rb]: Nice 💫

File: lib/nib/history.rb [hunk 3]
```diff
@@ -1,32 +1,62 @@
...
+      config_file.path
+    end
+
+    private
+
+    def config
+      if File.exist?(host_path)
+        File.read(host_path)
+      else
+        Nib.load_config(:shell, type)
```
Comment (binarymason) [inline lib/nib/history.rb]: Would this be more explicit if the method was `Nib.load_default_config`?

**Comment by johnallen3d** [comment]
If you want to try this out to see if it actually does what I claim you can run the following:

```sh
git clone git@github.com:technekes/nib.git
cd nib
git checkout feature/support-user-pry-irb-config
docker-compose run --rm gem rake build && gem uninstall -ax nib && gem install pkg/nib-1.4.1.gem
```

This will replace your currently installed version of `nib` with one built from the source of this PR/branch.

File: lib/nib/history.rb [hunk 4]
```diff
@@ -1,32 +1,62 @@
...
+      end
+    end
+
+    def config_file
+      @config_file ||= File.open("./tmp/#{type}", 'w+') do |file|
```
Comment (binarymason) [inline lib/nib/history.rb]: Why not use `FileUtils.cp` to copy the host file and then append the history command stuff?

File: lib/nib/history.rb (hunk 3 above)
Comment (johnallen3d) [inline lib/nib/history.rb]: Maybe in this case, but this method is [used elsewhere](https://github.com/technekes/nib/blob/16cc0bf07683fa0833c3a8677f437ea0e7ac6011/lib/nib/code_climate.rb#L27) and I don't think that name makes as much sense in that case. We could alias it for this use case if you think that would be better.

File: lib/nib/history.rb (hunk 4 above)
Comment (johnallen3d) [inline lib/nib/history.rb]: Because it's possible that the host doesn't define an `irbrc` or `pryrc`.

File: lib/nib/history.rb (hunk 3 above)
Comment (binarymason) [inline lib/nib/history.rb]: For a guy not accustomed to the codebase, I honestly was a little confused by `load_config`.  I thought that might mean to `load_config` from the host.  Definitely 👍 on making an alias.

File: lib/nib/history.rb (hunk 4 above)
Comment (binarymason) [inline lib/nib/history.rb]: Probably personal preference, I maybe would have `return unless File.exist?`.  But hey, it works! ✨ :shipit:

Comment (johnallen3d) [inline lib/nib/history.rb]: hmm the point is if the host does not have config then we can provide a "default". If we `return unless File.exist?` then we wouldn't end up writing out the default. Unless I'm misunderstanding. Can you a more complete example?

File: lib/nib/history.rb (hunk 3 above)
Comment (johnallen3d) [inline lib/nib/history.rb]: Alias added.

**Pull request merged**